cmake/package_tooling.cmake

```diff
@@ -0,0 +1,113 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2021, Nordic Semiconductor ASA
+
+# The Zephyr package tooling script provides a generic script mode interface
+# to the Zephyr CMake package and module structure.
+#
+# The purpose of this script is to provide a single entry for running any Zephyr
+# build tool that is executed during CMake configure time without creating a
+# complete build system.
+#
+# It does so by loading the Zephyr CMake modules specified with the 'MODULES' or
+# 'USAGE' argument.
+#
+# As basic set of modules are always loaded to ensure identical behavior when
+# running tooling identical to Zephyr CMake configure time.
+#
+# 'USAGE' mode:
+# This mode prints Zephyr build system usage for the given module.
+# Exactly what is printed for the specified module is module specific so consult
+# the documentation for the module itself.
+# For example, USAGE=boards will print a list of boards in the system:
+#   'cmake -DUSAGE=boards -P package_tooling.cmake'
+#
+# 'MODULES' mode:
+# This mode execute the given module identical to Zephyr CMake configure time.
+# When running 'MODULES' mode, then the application source directory must be
+# specified using '-DAPPLICATION_SOURCE_DIR=<path-to-sample>'
+#
+# The build directory will default to current working directory but can be
+# controlled with: '-DAPPLICATION_BINARY_DIR=<path-to-build-folder>'
+#
+# For example, if you were invoking CMake for 'hello_world' sample as:
+#   $ cd samples/hello_world/build
+#   $ cmake -DBOARD=<board> ..
+#
+#   you can invoke only dts module (and dependencies) as:
+#   $ cd samples/hello_world/build
```

Comment:
> I think this would be clearer if you added the line mkdir samples/hello_world/build before it.

Good point.
Will do.

>  these comments made me think I had to run the full cmake build system generation step before I could run this, which would defeat the purpose of the PR.

Actually my long term plan is to get this integrated with `west build` as an option, like:
`west build --only-dts -b<board> <sample>` or perhaps `west build --cmake-module=dts -b<board> <sample>`.
With such an approach, a  user can use the normal build flow, but ask the build system to process up to and including a specific module.

But I intent to do that separately from this PR.

Reply:
> > I think this would be clearer if you added the line mkdir samples/hello_world/build before it.
>
> Good point.
> Will do.

Done

Reply:
@mbolivar-nordic with latest support of `-B <path-to-build>` then the whole `mkdir build` and `cd build` can be removed.
This is more in line as what can be done directly with CMake.

Of course `mkdir` and `cd` is still possible, but we don't need to describe all various ways CMake can be invoked.